From fe2cd2e769b7d0b3af45d6d0a1e5e7f3a2b5cb87 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 29 Jan 2015 10:36:43 +0100
Subject: [PATCH 31/60] prefs: use system-provided preference skeleton

This patch makes it possible for distributions to have a global
default preferences file in /etc/iridium-browser/preferences that get
copied over to ~ when the user starts the program for the first time.
---
 chrome/browser/first_run/first_run_internal_linux.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/chrome/browser/first_run/first_run_internal_linux.cc b/chrome/browser/first_run/first_run_internal_linux.cc
index c67666610c39..40d86c4d6f2d 100644
--- a/chrome/browser/first_run/first_run_internal_linux.cc
+++ b/chrome/browser/first_run/first_run_internal_linux.cc
@@ -20,8 +20,12 @@ bool IsOrganicFirstRun() {
 base::FilePath MasterPrefsPath() {
   // The standard location of the master prefs is next to the chrome binary.
   base::FilePath master_prefs;
+#ifndef ANDROID
+  master_prefs = base::FilePath("/etc/iridium-browser");
+#else
   if (!base::PathService::Get(base::DIR_EXE, &master_prefs))
     return base::FilePath();
+#endif
   return master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 }
 
-- 
2.19.1

